### PR TITLE
Mark secret variables `sensitive`

### PR DESCRIPTION
**Pull request by jeffgran-dox**
I noticed that these password/authentication variables were not marked `sensitive` to protect them from being output into logs and console output.

### Test plan

- Ensure that `sensitive` is spelled right.
- Run automated tests to ensure terraform plan/apply can be run with this code.

### CODE REVIEW

**Review by Piszmog — APPROVED** [review]

**Review by eseliger — APPROVED** [review]

**Comment by Piszmog** [comment]
After this is merged, is this urgently needed or can it wait till that next Sourcegraph release? (we can't do a patch release, so no big deal if you need it)

**Pull request merged**
